test/lit/passes/dce-eh.wast

```diff
@@ -0,0 +1,145 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --dce -all -S -o - | filecheck %s
+
+;; If either try body or catch body is reachable, the whole try construct is
+;; reachable.
+(module
+  ;; CHECK:      (tag $e)
+  (tag $e)
+
+  ;; CHECK:      (tag $e-i32 (param i32))
+  (tag $e-i32 (param i32))
+
+  ;; CHECK:      (func $foo (type $0)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $foo)
+
+  ;; CHECK:      (func $try_unreachable (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch_all $catch)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $foo)
+  ;; CHECK-NEXT: )
+  (func $try_unreachable
+    (block $catch
+      (try_table (catch_all $catch)
+        (unreachable)
+      )
+    )
+    (call $foo) ;; shouldn't be dce'd
+  )
+
+  ;; CHECK:      (func $catch_unreachable (type $0)
+  ;; CHECK-NEXT:  (block $tryend
+  ;; CHECK-NEXT:   (block $catch
+  ;; CHECK-NEXT:    (try_table (catch_all $catch)
+  ;; CHECK-NEXT:     (br $tryend)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (unreachable)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $foo)
+  ;; CHECK-NEXT: )
+  (func $catch_unreachable
+    (block $tryend
+      (block $catch
+        (try_table (catch_all $catch)
+          (br $tryend)
+        )
+      )
+      (unreachable)
+    )
+    (call $foo) ;; shouldn't be dce'd
+  )
+
+  ;; CHECK:      (func $both_unreachable (type $0)
+  ;; CHECK-NEXT:  (block $tryend
+  ;; CHECK-NEXT:   (block $catch
+  ;; CHECK-NEXT:    (try_table (catch_all $catch)
+  ;; CHECK-NEXT:     (unreachable)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (unreachable)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $both_unreachable
+    (block $tryend
+      (block $catch
+        (try_table (catch_all $catch)
+          (unreachable)
+        )
+      )
+      (unreachable)
+    )
+    (call $foo) ;; should be dce'd
+  )
+
+  ;; CHECK:      (func $throw (type $0)
+  ;; CHECK-NEXT:  (block $label$0
+  ;; CHECK-NEXT:   (block $label$1
+  ;; CHECK-NEXT:    (throw $e)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $throw
```

Comment:
Given that this test doesn't contain any new instructions, I think we can delete this from `dce-eh-legacy.wast`, as we did for `code-pushing-eh-legacy.wast`.